django/contrib/staticfiles/finders.py

```diff
@@ -75,6 +75,9 @@ def check(self, **kwargs):
                 hint='Perhaps you forgot a trailing comma?',
                 id='staticfiles.E001',
             ))
+            if isinstance(settings.STATICFILES_DIRS, str):
+                # Iterating each character below would not be helpful.
+                return errors
```

Comment:
This is a separate cleanup, so please move it do a separate commit.

Reply:
Do we need to check that it's a string here? 

If it's not a list, a tuple, it seems unlikely to be some other iterable for which the further checks would be useful. As such - most likely an error will be raised when it's iterated over and the error message above won't even be displayed. (I'm imaging some people might use a Path instance).

This will then become a hard-to-diagnose problem for beginners, who will have a confusing exception rather than a useful error message from django. I wonder if for their sake it's worth always returning here.

Reply:
Yeah if a `dict` or `set` falls through, it will just crash on `os.path.abspath()`, so I agree further checks are extremely unlikely to happen anyway. I'll just return here.